docs/plugins/plugin-development-guide/implementation.rst

```diff
@@ -0,0 +1,216 @@
+Implementation
+--------------
+
+The goal of our plugin is simple - we need to extend the ``ProductVariant`` entity and provide a new flag, that could be set
+on the product variant form. Following customizations are done just like in the **Sylius Customization Guide**,
+take a look at :doc:`customizing models</customization/model>`, :doc:`form</customization/form>` and :doc:`template</customization/template>`.
+
+.. attention::
+
+    ``PluginSkeleton`` is focused on delivering the most friendly and testable environment. That's why in ``tests/Application`` directory,
+    there is a **tiny Sylius application** placed, with your plugin already used. Thanks to that, you can test your plugin with Behat scenarios
+    **within** Sylius application without installing it to any test app manually! There is, however, one important consequence of such an architecture.
+    **Everything** that should be done by a plugin user (configuration import, templates copying etc.) should also be done in ``tests/Application``
+    to simulate the real developer behavior - and therefore make your new features testable.
+
+Model
+*****
+
+The only field we need to add is an additional ``$availableOnDemand`` boolean. We should start with the unit tests (written with
+PHPSpec/PHPUnit/any other unit testing tool):
+
+.. code-block:: php
+
+    <?php
+
+    // spec/Entity/ProductVariantSpec.php
+
+    declare(strict_types=1);
+
+    namespace spec\IronMan\SyliusProductOnDemandPlugin\Entity;
+
+    use IronMan\SyliusProductOnDemandPlugin\Entity\ProductVariantInterface;
+    use PhpSpec\ObjectBehavior;
+    use Sylius\Component\Core\Model\ProductVariant;
+
+    final class ProductVariantSpec extends ObjectBehavior
+    {
+        function it_is_sylius_product_variant(): void
+        {
+            $this->shouldHaveType(ProductVariant::class);
+        }
+
+        function it_implements_product_variant_interface(): void
+        {
+            $this->shouldImplement(ProductVariantInterface::class);
+        }
+
+        function it_can_be_available_on_demand(): void
+        {
+            $this->setAvailableOnDemand(true);
+            $this->isAvailableOnDemand()->shouldReturn(true);
+        }
+    }
+
+.. code-block:: php
+
+    <?php
+
+    // src/Entity/ProductVariant.php
+
+    declare(strict_types=1);
+
+    namespace IronMan\SyliusProductOnDemandPlugin\Entity;
+
+    use Sylius\Component\Core\Model\ProductVariant as BaseProductVariant;
+
+    class ProductVariant extends BaseProductVariant implements ProductVariantInterface
+    {
+        /** @var bool */
+        private $availableOnDemand;
```

Comment:
What do you think about providing a default value for this property and removing nullable typehints? I think it makes sense for a boolean.